src/OneBlinkFormBase.tsx

```diff
@@ -532,21 +548,45 @@ function OneBlinkFormBase({
     [definition],
   )
 
+  const addCaptchaRef = React.useCallback((recaptcha: ReCAPTCHA) => {
+    captchasRef.current.push(recaptcha)
+    // this allows the FormElementCaptcha element to unregister any captchas
+    return () => {
+      captchasRef.current = captchasRef.current.filter(
+        (recaptchaInstance) => recaptchaInstance === recaptcha,
+      )
+    }
+  }, [])
+
+  const captchaContextValue = React.useMemo(
+    () => ({
+      captchaSiteKey,
+      captchaType: recaptchaType,
+      addCaptchaRef,
+    }),
+    [addCaptchaRef, captchaSiteKey, recaptchaType],
+  )
+
   const resetRecaptchas = React.useCallback(() => {
+    // unset the submission model value for each captcha element
     const updatedModel = { ...submission }
     formElementsService.forEachFormElement(definition.elements, (element) => {
       if (element.type === 'captcha') {
         updatedModel[element.name] = undefined
       }
     })
+    // reset each captcha
+    captchasRef.current.forEach((captcha) => {
+      captcha.reset()
+    })
     setHasAttemptedSubmit(false)
     setFormSubmission((current) => {
       return { ...current, submission: updatedModel }
     })
   }, [definition.elements, setFormSubmission, submission])
 
   const handleSubmit = React.useCallback(
-    (
+    async (
```

Comment:
If we are making this async, we need to: 
1. wrap in a try/catch to handle errors and also show some 
2. Disable the submit button while processing to prevent double clicking
3. show something to indicate something is happening in case the token(s) takes a long time to generate e.g. loading animation

Reply:
All settled should handle 1